sys/boot/i386/common/bootargs.h

```diff
@@ -64,10 +64,28 @@ struct bootargs
 	 */
 };
 
+#ifdef LOADER_GELI_SUPPORT
+#include <crypto/intake.h>
+#endif
+
 struct geli_boot_args
 {
     uint32_t		size;
-    char		gelipw[256];
+    union {
+        char            gelipw[256];
+        struct {
+            char                notapw;	/* 
+					 * single null byte to stop keybuf
+					 * being interpreted as a password
+					 */
+            uint32_t            keybuf_sentinel;
+#ifdef LOADER_GELI_SUPPORT
+            struct keybuf       *keybuf;
+#else
+            void                *keybuf;
```

Comment:
I did this so the struct will be the full size no matter what, but we don't require the definition of struct keybuf if it is not geliboot. I am not sure this is really useful